pkgs/applications/editors/jetbrains/plugins/default.nix

```
@@ -134,9 +134,9 @@ in
               fi
             done
 
-            for exe in $out/bin/${meta.mainProgram}*; do
-              if [[ -x "$exe" ]]; then
-                substituteInPlace $(realpath "$exe") --replace-warn '${ide.outPath}' $out
+            for exe in $out/${rootDir}/bin/*; do
+              if [ -x "$exe" ] && ( file "$exe" | grep -q 'text' ); then
+                substituteInPlace "$exe" --replace-quiet '${ide}' $out
```

```suggestion
                substituteInPlace "$exe" --replace-warn '${ide}' $out
```
warn should be the better choice, just in case anything goes wrong and we should know that it failed.

---

The problem is false positives; there are other scripts in this directory that don't contain the outPath

---

The ideal would be to know in advance which files should be substituted and use `--replace-fail` to ensure we notice if it changes.

If that can be easily constructed from the IDE name (etc), then great!

But for simplicity, and landing the fix quickly, I previously suggested just iterating over all executable text files with `--replace-quiet`.

---

Yep, I am just gonna go ahead and merge this. Would be nice to get those changes in a next PR.

---

> Would be nice to get those changes in a next PR.

I had a quick look. The path relies on `loName` (i.e. `toLower productShort`), and neither `loName` nor `productShort` are exposed as attributes on the IDE packages (AFAICT).

So I think calculating the paths explicitly would require some additional passthrus or other refactoring.

TBH I don't see it as worth the effort; replacing one `$out` path with another is analogous to a lot of "fixup" stage hooks, such as shebang patching. Such hooks simply fix all occurrences of a problem, without really caring if no problems are found.

Maybe it'd be worth moving from the `build` phase to the `fixup` phase, but that's subjective and nit-picky :grin: